lib/bamboo/test.ex

```diff
@@ -255,7 +267,7 @@ defmodule Bamboo.Test do
         receive do
           {:delivered_email, email} -> Map.from_struct(email)
         after
-          100 -> []
+          Bamboo.Test.refute_timeout() -> []
```

Comment:
If this is going to be a potentially test breaking concern is it worth a deprecation warning or minor version bump?

Reply:
Yeah, this is a good question. I honestly don't know. Technically, this is a bug that we're fixing, so it could be a patch. But if we consider the existing behavior as correct and us changing this, it could be considered potentially breaking. What do you think?

Reply:
Now that I've thought about this more. Maybe the change we should make is to default the `refute_timeout` to 100ms instead of 0ms. That would make this not a breaking change. Other tests might wait longer if the email fails to deliver, but they pass as soon as the email is received, so 100ms is a better default than 0ms. 

I also like that ExUnit's [refute_receive_timeout](https://hexdocs.pm/ex_unit/ExUnit.html#configure/1-options) defaults to 100ms. So it keeps us consistent with that, which probably matches people's expectations more closely.

Reply:
That seems like a nice approach to this. They can always change the default explicitly but half a second longer test suite seems like a low price to pay for a suitable default.

Reply:
@brimatteng I updated the default to 100ms, so there's no longer a potential for a breaking change, and I updated the PR description. 

I'm also hoping to quickly follow up with a PR that allows users to pass the `timeout` directly, so users should be able to change the timeout per test as desired.